examples/typescript/lnd/src/nodeHandler.ts

```diff
@@ -0,0 +1,77 @@
+import { LnRpc, WalletUnlockerRpc } from '@radar/lnrpc';
+import inquirer from 'inquirer';
+
+enum BaseAnswers {
+  OpenChannel = 'Open a channel',
+  ListPeers = 'List peers',
+  ListPendingChannels = 'List pending channels',
+  ListOpenChannels = 'List open channels',
+}
+
+type Base = {
+  action: BaseAnswers;
+};
+
+export default class NodeHandler {
+  constructor(private lnrpc: LnRpc & WalletUnlockerRpc) {}
+
+  whatWouldYouLikeToDo = async () => {
+    const info = await this.lnrpc.getInfo();
+    const { action } = await inquirer.prompt<Base>({
+      type: 'list',
+      name: 'action',
+      message: `Hi ${info.alias}, what would you like to do?`,
+      choices: Object.values(BaseAnswers),
+    });
+    let answer = await this.getAnswer(action);
+    while (typeof answer !== 'string') {
+      answer = await answer();
+    }
+    console.log(answer);
+    await this.whatWouldYouLikeToDo();
+  };
+
+  getAnswer = async (
+    question: BaseAnswers,
+  ): Promise<string | (() => Promise<string>)> => {
+    switch (question) {
+      case 'Open a channel': {
+        return this.specifyChannelOpeningParameters;
+      }
+      case 'List peers': {
+        const { peers } = await this.lnrpc.listPeers();
+        return `Your peers:\n\n`.concat(peers.map(p => p.address).join('\n'));
+      }
+      case 'List pending channels': {
+        const info = await this.lnrpc.pendingChannels();
+        return `Pending channels:\n${JSON.stringify(info)}`;
+      }
+      case 'List open channels': {
+        const info = await this.lnrpc.listChannels();
+        return `Open channels:\n${JSON.stringify(info)}`;
+      }
+      default:
+        throw new Error('Question not recognized');
+    }
+  };
+
+  specifyChannelOpeningParameters = async () => {
```

Comment:
@jamaljsr I haven't quite tidied the PR up, but I added the discuss command to open a channel with a peer. It seems to work fine, but I'm not familiar with lightning development so I don't know what I don't know. Could you take a look at this method and let me know if there are other things that you think need to be added or changed here?

Reply:
This looks good to me 👍